### PR TITLE
 Fix a Augmented Elekk issue #3760

### PR DESCRIPTION
**Pull request by junsix**
- [x] I have read the [Contributing Guidelines](https://github.com/HearthSim/Hearthstone-Deck-Tracker/blob/master/CONTRIBUTING.md#contributing)
- [x] I have already signed the CLA 

Now, if  Augmented Elekk is played, the deck tracker count the cards created by Augmented Elekk.

### CODE REVIEW

File: Hearthstone Deck Tracker/LogReader/HsGameState.cs [hunk 1]
```diff
@@ -34,6 +34,7 @@ public HsGameState(GameV2 game)
 		public int JoustReveals { get; set; }
 		public Dictionary<int, IList<string>> KnownCardIds { get; set; }
 		public int LastCardPlayed { get; set; }
+		public String LastAddedCardId { get; set; }
```
Comment (riQQ) [inline Hearthstone Deck Tracker/LogReader/HsGameState.cs]:
```suggestion
		public string LastAddedCardId { get; set; }
```

File: Hearthstone Deck Tracker/LogReader/Interfaces/IHsGameState.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@ public interface IHsGameState
 		int JoustReveals { get; set; }
 		Dictionary<int, IList<string>> KnownCardIds { get; set; }
 		int LastCardPlayed { get; set; }
+		String LastAddedCardId { get; set; }
```
Comment (riQQ) [inline Hearthstone Deck Tracker/LogReader/Interfaces/IHsGameState.cs]:
```suggestion
		string LastAddedCardId { get; set; }
```

File: Hearthstone Deck Tracker/LogReader/Handlers/PowerHandler.cs [hunk 3]
```diff
@@ -138,6 +138,19 @@ public void Handle(string logLine, IHsGameState gameState, IGame game)
 								Log.Info($"Found known cardId for entity {id}: {cardId}");
 								gameState.KnownCardIds[blockId.Value].Remove(cardId);
 							}
+							gameState.LastAddedCardId = cardId;
+						}
+						else if(blockId.HasValue && !gameState.KnownCardIds.ContainsKey(blockId.Value))
+						{
+							//Augmented Elekk
+							if(gameState.CurrentBlock.CardId == "BOT_559")
```
Comment (riQQ) [inline Hearthstone Deck Tracker/LogReader/Handlers/PowerHandler.cs]:
```suggestion
							if(gameState.CurrentBlock.CardId == Collectible.Neutral.AugmentedElekk)
```

Comment (azeier) [inline Hearthstone Deck Tracker/LogReader/Handlers/PowerHandler.cs]:
```suggestion
							if(gameState.CurrentBlock.CardId == Collectible.Neutral.AugmentedElekk)
```

File: Hearthstone Deck Tracker/LogReader/Handlers/PowerHandler.cs [hunk 4]
```diff
@@ -138,6 +138,19 @@ public void Handle(string logLine, IHsGameState gameState, IGame game)
 								Log.Info($"Found known cardId for entity {id}: {cardId}");
 								gameState.KnownCardIds[blockId.Value].Remove(cardId);
 							}
+							gameState.LastAddedCardId = cardId;
+						}
+						else if(blockId.HasValue && !gameState.KnownCardIds.ContainsKey(blockId.Value))
+						{
+							//Augmented Elekk
```
Comment (azeier) [inline Hearthstone Deck Tracker/LogReader/Handlers/PowerHandler.cs]:
```suggestion
```

Comment (riQQ) [inline Hearthstone Deck Tracker/LogReader/Handlers/PowerHandler.cs]:
```suggestion
```

**Comment by junsix** [comment]
Thank you for your comments.

**Comment by riQQ** [comment]
I took a look at the power log and its reduced form looks like this:
```
BLOCK_START BlockType=PLAY Fal'dorei Strider
	BLOCK_START BlockType=POWER Fal'dorei Strider
		Shuffle 3
	BLOCK_END
	BLOCK_START BlockType=TRIGGER Augmented Elekk
		Shuffle 1
	BLOCK_END
	BLOCK_START BlockType=TRIGGER Augmented Elekk
		Shuffle 1
	BLOCK_END
	BLOCK_START BlockType=TRIGGER Augmented Elekk
		Shuffle 1
	BLOCK_END
BLOCK_END
```

This means, that the property would have to be in the parent block. The question left open is what does it look like when cards are shuffled by other means e.g. `Academic Espionage` (a spell).

**Comment by junsix** [comment]
Thank you for your kind answer. I move the LastAddedCardId variable to Block.

**Comment by azeier** [comment]
Thanks for the fix!

And thanks for reviewing it @riQQ 😃 .

**Pull request merged**